### PR TITLE
[None][chore] Drop the skip_* sampling flags from one-model spec metadata

### PR DESCRIPTION
**Pull request by zhaoyangwang-nvidia**
## Description

`SpecMetadata` carried `skip_temperature` / `skip_top_k` / `skip_top_p`, recomputed each step from whether any request in the batch used that filter, alongside `AdvancedSamplingMode`, which expresses the same "disable this filter" intent as deploy-time configuration. The two were introduced independently and never reconciled. This PR removes the `skip_*` flags.

- `skip_temperature` had no reader left. It was a parameter of the sampling op until the FlashInfer sampler refactors (#15542, #16160) dropped it from the signature, and the assignment in `_scan_one_model_sampling` was left behind.
- `skip_top_k` / `skip_top_p` had one reader, in the rejection path, where they selected between passing a tensor and passing `None`. That choice is made in Python, so CUDA graph capture bakes it in, and unlike `is_all_greedy_sample` these flags are not part of the graph key: a replayed graph keeps whichever variant capture happened to see, so the per-batch adaptivity the flags suggest does not survive once graphs are enabled. The capture-time override (`_force_non_greedy_for_capture`) having to force them to `False` is the same observation from the other side — if they were live at replay, the override would be unnecessary.
- `AdvancedSamplingMode` covers the same four sampling sites and, being fixed for the deployment, has no such gap.

Under the default `FULL` mode a request that sets `top_k` / `top_p` now always gets that filter applied. Previously a batch where no request used a filter would skip its kernel, subject to the capture caveat above; a deployment that wants those kernels skipped selects `NO_TOPK` / `NO_TOPP` / `NO_TOPK_NO_TOPP` explicitly. On the eager path this costs two extra kernel launches (~7 us) for batches that use neither filter. Under CUDA graphs there is no change, since the captured graph already contained whatever the capture-time flags selected.

`is_all_greedy_sample` is unaffected: it is derived from per-request greediness, not from which filters are in use, and stays in the graph key.

Not addressed here: `AdvancedSamplingMode` is applied at four sampling sites in `interface.py`, but three further sampling calls bypass it — `eagle3_dynamic_tree.py:775` and `:883`, and `dynamic_tree_ops.py:271` — so a deployment using dynamic tree does not get the filter kernels skipped there. `skip_*` never covered those sites either, so this PR neither improves nor regresses them. Worth a follow-up.

## Test Coverage

Validated on H200 NVL (SM 9.0).

- `test_capture_override_leak.py` passes (4 cases). It covers the capture-time override path these flags participated in; the three `skip_*` assertions are dropped and the `is_all_greedy_sample` assertion, the load-bearing one, stays.
- End-to-end `test_llama_eagle3_rejection_sampling_modes` passes (4 cases), covering `use_cuda_graph` x `use_dynamic_tree` with `temperature=1.0` / `top_p=0.9` / `top_k=50` so the non-greedy path is exercised.
- A repository-wide grep confirms no remaining references to the removed flags.
- Changed-file pre-commit hooks pass.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/speculative/interface.py [hunk 1]
```diff
@@ -1745,15 +1720,13 @@ def _sample_and_accept_draft_tokens_rejection(
...
-            top_ps = (None if spec_metadata.skip_top_p else
-                      spec_metadata.top_ps[gen_start:gen_end])
+            # A filter the mode disables becomes None, which lets the C++ op
+            # short-circuit on a host-side check rather than an
+            # `.item<bool>()` sync that would break CUDA graph capture.
             top_ks, top_ps = resolve_advanced_sampling_filters(
-                spec_metadata.advanced_sampling_mode, top_ks, top_ps)
+                spec_metadata.advanced_sampling_mode,
+                spec_metadata.top_ks[gen_start:gen_end],
+                spec_metadata.top_ps[gen_start:gen_end])
```
[the remark this answers is not in the document]
Comment (zhaoyangwang-nvidia) [reply tensorrt_llm/_torch/speculative/interface.py:1729]: Checked this three ways; I don't think it applies here.

**No alternate implementation exists.** `compute_probs_from_logits_op` is a straight line — `top_k_mask_logits` (if `top_k`), `softmax`, `top_p_renorm_probs` (if `top_p`). A `None` `top_p` skips the renorm step; it does not select a different probability path.

**Measured on H200, `NO_TOPP` and `NO_TOPK_NO_TOPP`, 16x4096 logits:** `None` vs an explicit `top_p=1.0` tensor differ by max 1.8e-07 (float32 rounding on the renorm), and `argmax` is identical. Control: a real `top_p=0.9` against `None` differs by 8.4e-02, so the comparison does detect a live filter. `top_p=1.0` renorm is an identity — it keeps the full probability mass and renormalizes to the same distribution — so skipping it is token-equivalent by construction, not coincidence.

**This PR doesn't change that behaviour.** The `resolve_advanced_sampling_filters(mode, ...)` call and its `skips_top_p -> None` semantics are unchanged from the base commit (introduced in #16561); this PR only removes the `skip_*` flags upstream of it.

Also, carrying a `top_p=1.0` tensor instead of `None` would defeat the point of `NO_TOPP`, which exists precisely to skip that kernel.

Your other suggestion is fair though: `test_advanced_sampling_mode.py` covers `NO_TOPK` vs `FULL` distribution equivalence but not `NO_TOPP` / `NO_TOPK_NO_TOPP`. That's a gap in the `AdvancedSamplingMode` tests rather than something this PR introduces — tracking it as a follow-up.
